### PR TITLE
udpate docker image to use rustup

### PR DESCRIPTION
**Pull request by Ragnaroek**
A clean version of the previous pull request (I did the merge after introducing the changes, that's why it did show three commits in the previous PR).

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,62 +1,43 @@
...
+ENV HOME=/home/cross
+ENV URL_GIT_PI_TOOLS=https://github.com/raspberrypi/tools.git \
+    TOOLCHAIN_64=$HOME/pi-tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/bin \
+    TOOLCHAIN_32=$HOME/pi-tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian/bin
 
-ENV TOOLCHAIN_64=$HOME/pi-tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/bin \
-TOOLCHAIN_32=$HOME/pi-tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian/bin
+# install rustup with raspberry target
+RUN curl https://sh.rustup.rs -sSf > $HOME/install_rustup.sh
+RUN chmod +x $HOME/install_rustup.sh
```
Comment (Ogeon) [inline docker/Dockerfile]: Is this really necessary if it's running with `sh`?

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,62 +1,43 @@
...
-wget -qO- $URL_CARGO_ARCHIVE | tar xvz -C $HOME; \
-mkdir $HOME/.cargo;
+# configure cargo
 COPY conf/cargo-config $HOME/.cargo/config
 
 COPY bin $HOME/bin
 ENV PATH=$HOME/bin:$PATH
-
 ENTRYPOINT ["run.sh"]
-CMD ["help"]
```
Comment (Ogeon) [inline docker/Dockerfile:62]: Why was this removed?

File: docker/conf/cargo-config [hunk 3]
```diff
@@ -1,3 +1,3 @@
 [target.arm-unknown-linux-gnueabihf]
-ar = "arm-linux-gnueabihf-gcc-ar"
+ar = "arm-linux-gnueabihf-ar"
```
Comment (Ogeon) [inline docker/conf/cargo-config:2]: I suppose this is a symlink to the previous one, or did they rename it?

**Comment by Ogeon** [comment]
You could probably have just rebased it, I think, but this is fine. :smile: It looks really good, apart from a few details.

File: docker/Dockerfile (hunk 2 above)
Comment (Ragnaroek) [inline docker/Dockerfile:62]: i started the dockerfile from scratch, forgot this.

File: docker/conf/cargo-config (hunk 3 above)
Comment (Ragnaroek) [inline docker/conf/cargo-config:2]: I had some problems executing the linker and compared it with your solution here:
https://github.com/Ogeon/rust-wiringpi/blob/master/scripts/config

but either work

File: docker/Dockerfile (hunk 1 above)
Comment (Ragnaroek) [inline docker/Dockerfile]: no it's not, removed it

**Comment by Ragnaroek** [comment]
updated the commit, thx for the review

**Comment by Ogeon** [comment]
I think it would be nice to be able to choose Rust version. I can imagine some people wanting to use nightly, for example.

**Comment by Ragnaroek** [comment]
I agree, but we could add this feature in a later patch?
This would enable using rust 1.10 again

**Comment by Ogeon** [comment]
Sure, I just thought it would be relatively simple to add. Just check if something like `RUST_VERSION` is set, and then switch over to that version if it is. Just when installing.

**Comment by Ragnaroek** [comment]
well you were right, it was pretty simple (see latest commit that adds this feature)

**Comment by Ogeon** [comment]
Nice! Even simpler than I thought. Thank you for putting up with my request and thank you for doing these changes!
